graphstorm-processing/scripts/run_repartitioning.py

```diff
@@ -0,0 +1,174 @@
+You may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Run repartitioning job on processed data on SageMaker.
+
+This script allows us to launch a SageMaker Processing job to
+repartition the data produced by the graph processing job.
+
+Usage:
+    run_repartitioning.py --input-prefix <s3--prefix> \\
+        --config-file metadata.json --image-uri <image-uri> --role <role>
+
+Script Parameters
+----------
+Required:
+
+--s3-input-prefix: str
+    S3 path prefix to the input data.
+--config-filename: str
+    Distributed partitioning metadata JSON file. Created by the processing job,
+    the default name should be `metadata.json`.
+--image-uri: str
+    ECR image URI for the GSProcessing container. Should be in the form:
+    ``ACCOUNT_ID.dkr.ecr.REGION.amazonaws.com/graphstorm-processing:VERSION``
+--role: str
+    ARN of the SageMaker execution IAM role to use for the processing job.
+    Should be in the form: `a`rn:aws:iam::ACCOUNT_ID:role/ROLE_NAME``
+
+Optional:
+--region: str
+    AWS region for the processing job. If not provided will use the input bucket's region.
+--instance-type: str
+    EC2 instance type for the processing job. (Default: 'ml.r5.4xlarge')
+--job-name: str
+    Prefix name for the processing job. (Default: 'gsprocessing-repartitioning')
+--container-log-level: str
+    Log level for the container. (Default: INFO)
+--host-log-level: str
+    Log level for the host. (Default: INFO)
+--sm-processor-parameters
+    Parameters to be passed to the SageMaker Estimator as kwargs,
+    in <key>=<val> format, separated by spaces.
+    When passing lists or other objects, do not include spaces in the <val>, e.g.
+    ``--sm-estimator-parameters \"volume_size=100 subnets=['subnet-123','subnet-345']
+    security_group_ids=['sg-1234','sg-3456']\"``
+
+"""
+import argparse
+import logging
+from time import strftime, gmtime
+
+import boto3
+from botocore.config import Config
+import sagemaker
+from sagemaker.processing import Processor
+
+import script_utils  # pylint: disable=wrong-import-order
+
+
+def parse_args() -> argparse.Namespace:
+    """Parse repartitioning args"""
+    parser = script_utils.get_common_parser()  # type: argparse.ArgumentParser
+
+    return parser.parse_args()
+
+
+def main():
+    """Main entry point for re-partitioning."""
+    args = parse_args()
+    logging.basicConfig(level=args.host_log_level)
+    # Remove trailing slash from s3_input_prefix
+    raw_prefix = args.s3_input_prefix
+    s3_input_prefix = raw_prefix[:-1] if raw_prefix[-1] == "/" else raw_prefix
```

Comment:
Can we use functions in s3_utils to do this?

Reply:
That would require us to import gs_processing and all its dependencies to launch a job, we're trying to keep the dependencies of our launch scripts to just boto3 and sagemaker.